homeassistant/components/home_connect/__init__.py

```diff
@@ -315,3 +316,29 @@ def get_dict_from_home_connect_error(err: api.HomeConnectError) -> dict[str, Any
         if len(err.args) > 0 and isinstance(err.args[0], str)
         else {}
     )
+
+
+RE_CAMEL_CASE = re.compile(r"(?<!^)(?=[A-Z])")
+
+
+def bsh_key_to_translation_key(bsh_key: str) -> str:
+    """Convert a BSH key to a translation key format.
+
+    This function takes a BSH key, such as `Dishcare.Dishwasher.Program.Eco50`,
+    and converts it to a translation key format, such as `dishcare_dishwasher_bsh_key_eco50`.
+    """
+    return "-".join(
+        RE_CAMEL_CASE.sub("_", split) for split in bsh_key.split(".")
+    ).lower()
```

Comment:
Isn't this the same as `slugify()`?

Reply:
slugify transforms `HelloWorld` into `helloworld`